libvmi/core.c

```diff
@@ -77,22 +77,26 @@ open_config_file(
     dbprint(VMI_DEBUG_CORE, "--looking for config file at %s\n", location);
 
     f = fopen(location, "r");
-    free(location);
 
     if (f) {
         goto success;
     }
+    free(location);
 
     /* finally check in /etc */
     dbprint(VMI_DEBUG_CORE, "--looking for config file at /etc/libvmi.conf\n");
-    f = fopen("/etc/libvmi.conf", "r");
+    location = g_malloc0(strlen("/etc/libvmi.conf")+1);
```

Comment:
Matching g_free with g_malloc provides better portability and allows one to benefit from some memory tracking features for debugging memory leaks more easily.